### PR TITLE
Jib rpi visuals

### PR DESCRIPTION
**Pull request by jib**
Basic JosPack viz plug ins. Note this includes the changes from https://github.com/rjmcdougall/BB/pull/170 as without them, profiles could not be downloaded/created/updated.

### CODE REVIEW

**Comment by jib** [comment]
@rjmcdougall if you could review & merge this one, that'd be lovely

File: bb/src/main/java/com/richardmcdougall/bb/BoardVisualization.java [hunk 1]
```diff
@@ -284,12 +293,41 @@ public int displayAlgorithm(String algorithm) {
...
+                break;
+            // End JosPack visualizations
+
             case "modePlayaMap()":
                 frameRate = mFrameRate;
                 mVisualizationPlayaMap.update(Visualization.kDefault);
                 break;
 
             default:
+                Log.d(TAG, "Could not find visualization algorithm: " + algorithm);
```
Comment (rjmcdougall) [inline bb/src/main/java/com/richardmcdougall/bb/BoardVisualization.java]: We can't log.d every frame -- if we ever hit it it causes hang side effects on android. Maybe just print once?

Comment (jib) [reply bb/src/main/java/com/richardmcdougall/bb/BoardVisualization.java]: Fair enough - not sure how to print only once without keeping a status bit, so for now, we'll leave it commented out.

**Comment by jib** [comment]
@rjmcdougall - updated & merged with current master.

**Pull request merged**